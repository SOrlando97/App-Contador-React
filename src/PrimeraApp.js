import React from 'react'
import PropTypes from 'prop-types'
//import React, { Fragment } from 'react'
//funtional Compount

// const PrimeraApp = () =>{    

//     return (
//         <Fragment>
//             <h1>Hola Mundo </h1>
//             <p>Mi primera aplicaci√≥n</p>
//         </Fragment>        
//     );     
// }
//este hola mundo es un valor por defecto
const PrimeraApp = ({ saludo, subtitulo}) =>{    

    

    return (
        <>
            <h1>{saludo}</h1>
            {/* <pre>{ JSON.stringify(saludo,null,3) }</pre> */}
            <p>{subtitulo}</p>
        </>        
    );     
}
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
PrimeraApp.defaultProps ={
    subtitulo: 'Soy un Subtitulo'
}
export default PrimeraApp;