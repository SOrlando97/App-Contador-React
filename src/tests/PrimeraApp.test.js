import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import {shallow} from 'enzyme'
import '@testing-library/jest-dom';

describe('Prueba en PrimeraApp', ()=>{
    //esto con //import '@testing-library/jest-dom/extend-expect'; en setuptests
    // test('debe mostrar hola que hace', () => { 
        
    //     const saludo = 'Hola que hace';
    //     const {getByText} = render(<PrimeraApp saludo ={ saludo }/>)

    //     expect(getByText(saludo)).toBeInTheDocument();
    //  })
    test('debe de mostrar <PrimeraApp /> correctamente',()=>{
        const saludo = 'Hola que hace';
        const wrapper = shallow( <PrimeraApp saludo={saludo}/> );

        expect(wrapper).toMatchSnapshot();
    })
    test('debe de mostrar el subtitulo enviado por props', () => { 

        const saludo = 'Hola que hace';
        const subtitulo = 'Subtitulitos';
        const wrapper = shallow( 
        <PrimeraApp 
            saludo={saludo} 
            subtitulo={ subtitulo }
            /> 
        );

        const textoParrafo = wrapper.find('p').text();
        expect( textoParrafo ).toBe (subtitulo);

     })
})