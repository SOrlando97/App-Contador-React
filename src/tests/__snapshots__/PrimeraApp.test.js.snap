// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en PrimeraApp debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola que hace
     
  </h1>
  <p>
    Soy un Subtitulo
  </p>
</Fragment>
`;
